api/handlers/userhandler.py

```diff
@@ -23,7 +23,12 @@ def __init__(self, request=None, response=None):
     def get(self, _id):
         user = self._get_user(_id)
         permchecker = userauth.default(self, user)
-        result = permchecker(self.storage.exec_op)('GET', _id, projection={'api_key': 0} or None)
+        projection = {'api_key': 0}
+        if self.user_is_admin:
+            projection['wechat.openid'] = 0
+        else:
+            projection['wechat'] = 0
```

Comment:
Nope. I don't like that we have to explicitly call out wechat in the userhandler methods but it was the simplest way to solve the current problem. A more robust solution (if we support more auth types that need registration) is having a more generic `registration` map.